src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -608,7 +629,18 @@ private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool
                             if (_lexer.TextWindow.PeekChar(1) == '"' && !RecoveringFromRunawayLexing())
                             {
                                 // check for verbatim string inside an expression hole.
-                                ScanInterpolatedStringLiteralNestedVerbatimString();
+                                var nestedStringPosition = _lexer.TextWindow.Position;
+
+                                // Note that this verbatim string may encounter an error (for example if it contains a
+                                // new line and we don't allow that).  This should be reported to the user, but should
+                                // not put us into an unrecoverable position.  We can clearly see that this was intended
+                                // to be a normal verbatim string, so we can continue on an attempt to understand the
+                                // outer interpolated string properly.
+                                var discarded = default(TokenInfo);
+                                var errorCode = _lexer.ScanVerbatimStringLiteral(ref discarded, _allowNewlines);
+                                if (errorCode != null)
```

Comment:
>!= null [](http://example.com/codeflow?start=46&length=7)

Could be `errorCode is ErrorCode value` to avoid the double null-check in Value.